scripts/cmake/config/bfconfig_template.cmake

```diff
@@ -0,0 +1,41 @@
+#
+# Bareflank Hypervisor
+# Copyright (C) 2015 Assured Information Security, Inc.
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+# ------------------------------------------------------------------------------
+# Bareflank Build Configurations
+# ------------------------------------------------------------------------------
+
+# set(BUILD_TYPE Debug)
```

Comment:
We should provide documentation in the template so that all you have to do is read and modify the template.

Reply:
How much documentation should go in the template? I choose the commented out lines in the template as specific examples of 90% of what you would want to do with this file, namely:

* Setting a string type config
* Setting a boolean type config
* Adding an extension from GitHub
* Adding an extension from a local directory

What else should be added? Maybe a link or reference to the build instructions readme? One-sentence descriptions of what each section is for?

Reply:
I am a fan of redundancy, so in this case I would say adding the build instructions, and then a link to all of the options that are available in another doc would be fine so long as each option is documented somewhere.

Reply:
How about adding something like this:

```
# For instructions on configuring and building Bareflank, see hypervisor/scripts/docs/build_instructions.md

# To view all available configurations, use ccmake or cmake-gui from your build directory:
#      cd </path/to>/build; ccmake .
```

I've been avoiding creating a file that documents all available build configurations because those things are going to change, get added, and get dropped really fluidly as we continue development. As the build README indicates, the best way to view all of the available configs is to use ccmake or cmake-gui. From those, you can see every available option, the possible values for each, and a description for each. If we go the README route for build configs, we'll have to remember to add/modify two things every time we touch a build config.